clients/src/main/java/org/apache/kafka/clients/consumer/internals/CommitRequestManager.java

```diff
@@ -780,6 +981,28 @@ List<NetworkClientDelegate.UnsentRequest> drain(final long currentTimeMs) {
             return Collections.unmodifiableList(unsentRequests);
         }
 
+        /**
+         * Find the unsent commit requests that have expired, remove them and complete their
+         * futures with a TimeoutException.
+         */
+        private void failAndRemoveExpiredCommitRequests(final long currentTimeMs) {
+            List<OffsetCommitRequestState> expiredRequests = unsentOffsetCommits.stream()
+                .filter(req -> req.isExpired(currentTimeMs)).collect(Collectors.toList());
+            unsentOffsetCommits.removeAll(expiredRequests);
```

Comment:
This seems inefficient. Would it be better to use an iterator and to remove with it?

Reply:
Agree. Actually even a lot simpler with just `removeIf` and a `maybeExpire`. Done.